django_socio_grpc/services/servicer_proxy.py

```diff
@@ -279,23 +282,41 @@ def __getattr__(self, action):
 
         return self.get_handler(action)
 
-    def process_exception(self, exc, request_container: GRPCRequestContainer):
-        if isinstance(exc, GRPCException):
-            request_container.context.abort(exc.status_code, exc.get_full_details())
+    def _get_status_code_and_details(self, exc: Exception) -> Tuple[grpc.StatusCode, str]:
+        if isinstance(exc, APIException):
+            status_code = exc.status_code
+            if not isinstance(status_code, grpc.StatusCode):
+                status_code = HTTP_CODE_TO_GRPC_CODE.get(status_code, grpc.StatusCode.UNKNOWN)
+            return status_code, json.dumps(exc.get_full_details())
         else:
             details = type(exc).__name__
             if settings.DEBUG:
                 details = str(exc)
-            request_container.context.abort(grpc.StatusCode.UNKNOWN, details)
+            return grpc.StatusCode.UNKNOWN, details
 
-    async def async_process_exception(self, exc, context):
-        if isinstance(exc, GRPCException):
-            await context.abort(exc.status_code, exc.get_full_details())
-        else:
-            details = type(exc).__name__
-            if settings.DEBUG:
-                details = str(exc)
-            await context.abort(grpc.StatusCode.UNKNOWN, details)
+    def process_exception(self, exc: Exception, context: grpc.ServicerContext):
+        status_code, details = self._get_status_code_and_details(exc)
+        context.abort(status_code, details)
+
+    async def async_process_exception(self, exc: Exception, context: grpc.aio.ServicerContext):
+        status_code, details = self._get_status_code_and_details(exc)
+        await context.abort(status_code, details)
+
+    def log_exception(self, exception: Exception, message: str, extra={}):
```

Comment:
this need doc string and may not have it's place in servicer proxy. Maybe a method in exception.py too